coordinator/persistence-cassandra-5.0/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -0,0 +1,621 @@
+            preparedWithInfo.isUseKeyspace());
+    }
+    throw new ProtocolException("Unexpected type for RESULT message: " + resultMessage.kind);
+  }
+
+  public static PersistenceException convertInternalException(Throwable t) {
+    if (t instanceof CassandraException) {
+      return Conversion.toExternal((CassandraException) t);
+    } else if (t instanceof org.apache.cassandra.transport.ProtocolException) {
+      // Note that ProtocolException is not a CassandraException
+      org.apache.cassandra.transport.ProtocolException ex =
+          (org.apache.cassandra.transport.ProtocolException) t;
+      return new ProtocolException(t.getMessage(), toExternal(ex.getForcedProtocolVersion()));
+    } else if (t instanceof org.apache.cassandra.transport.ServerError) {
+      // Nor is ServerError
+      return new ServerError(t.getMessage());
+    }
+    return new ServerError(t);
+  }
+
+  public static BatchStatement.Type toInternal(BatchType external) {
+    switch (external) {
+      case LOGGED:
+        return BatchStatement.Type.LOGGED;
+      case UNLOGGED:
+        return BatchStatement.Type.UNLOGGED;
+      case COUNTER:
+        return BatchStatement.Type.COUNTER;
+    }
+    throw new IllegalArgumentException("Invalid batch type");
+  }
+
+  public static Column.ColumnType getTypeFromInternal(AbstractType abstractType) {
+    if (abstractType instanceof ReversedType) {
+      return getTypeFromInternal(((ReversedType) abstractType).baseType);
+    }
+    if (abstractType instanceof MapType) {
+      return Column.Type.Map.of(
+              getTypeFromInternal(((MapType) abstractType).getKeysType()),
+              getTypeFromInternal(((MapType) abstractType).getValuesType()))
+          .frozen(!abstractType.isMultiCell());
+    } else if (abstractType instanceof SetType) {
+      return Column.Type.Set.of(getTypeFromInternal(((SetType) abstractType).getElementsType()))
+          .frozen(!abstractType.isMultiCell());
+    } else if (abstractType instanceof ListType) {
+      return Column.Type.List.of(getTypeFromInternal(((ListType) abstractType).getElementsType()))
+          .frozen(!abstractType.isMultiCell());
+    } else if (abstractType.getClass().equals(TupleType.class)) {
+      TupleType tupleType = ((TupleType) abstractType);
+      return Column.Type.Tuple.of(
+              ((TupleType) abstractType)
+                  .allTypes().stream()
+                      .map(t -> getTypeFromInternal(t))
+                      .toArray(Column.ColumnType[]::new))
+          .frozen(!tupleType.isMultiCell());
+    } else if (abstractType.getClass().equals(UserType.class)) {
+      UserType udt = (UserType) abstractType;
+      return ImmutableUserDefinedType.builder()
+          .keyspace(udt.keyspace)
+          .name(udt.getNameAsString())
+          .addAllColumns(getUDTColumns(udt))
+          .build()
+          .frozen(!udt.isMultiCell());
+    } else if (abstractType.getClass().equals(VectorType.class)) {
```

Comment:
@maheshrajamani So I think handling happens here, to pass parameterization necessary.